### PR TITLE
ignore error parsing container number label, just warn

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
don't throw an error as we can't retrieve or parse container number label, just warn
While it's unclear how we end in such a bad state, this should not break compose usage as this has limited impact - we only use this to suffix container name with a unique number, nothing critical

**Related issue**
fixes https://github.com/docker/compose/issues/10004

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

**Review by glours — APPROVED** [review]
LGTM

File: pkg/compose/convergence.go [hunk 1]
```diff
@@ -370,18 +367,23 @@ func shouldWaitForDependency(serviceName string, dependencyConfig types.ServiceD
...
 }
 
-func nextContainerNumber(containers []moby.Container) (int, error) {
+func nextContainerNumber(containers []moby.Container) int {
 	max := 0
 	for _, c := range containers {
-		n, err := strconv.Atoi(c.Labels[api.ContainerNumberLabel])
+		s, ok := c.Labels[api.ContainerNumberLabel]
+		if !ok {
+			logrus.Warnf("container %q is missing %s label", c.ID, api.ContainerNumberLabel)
```
Comment (ulyssessouza) [inline pkg/compose/convergence.go:375]: nit:
```suggestion
			logrus.Warnf("container %q is missing %q label", c.ID, api.ContainerNumberLabel)
```

**Review by ulyssessouza — APPROVED** [review]
LGTM!
Just nits on %s -> %q

**Comment by nickform-ost** [comment]
I'm on macOS. The dashboard app shows version number v4.14.1 and I still hit the bug. Any idea how to estimate when this will make it all the way to users? Also, are you in a position to suggest any workaround that could help us out in the meantime?

**Comment by glours** [comment]
@nickform-ost next release of Docker Desktop should be release soon with the `2.13.0` version of Compose
If you can, a `docker system prune -af` should resolve the issue

**Comment by nickform-ost** [comment]
I really appreciate you getting back to me. That seems to have done the trick!

**Pull request merged**